src/kern/npf_ctl.c

```diff
@@ -732,6 +755,63 @@ npfctl_save(npf_t *npf, u_long cmd, void *data)
 	return error;
 }
 
+/*
+ * npfctl_table_replace_nvlist: atomically replace a table's contents
+ * with the passed table data.
+ */
+static int __noinline
+npfctl_table_replace_nvlist(npf_t *npf, nvlist_t *npf_dict, nvlist_t *errdict)
+{
+	npf_tableset_t *tblset;
+	int error = 0;
+
+	npf_table_t *tbl, *gc_tbl;
```

Comment:
Please remove the trailing line above. Also, consider this:
```c
	npf_table_t *tbl, *gc_tbl = NULL;
	npf_tableset_t *tblset;
	int error = 0;
```
And then instead of another `goto`:
```c
	...
	npf_config_sync(npf);
err:
	npf_config_exit(npf);
	if (gc_tbl) {
		npf_table_destroy(gc_tbl);
	}
	return error;
}
```

Reply:
I've done it this way now. Sorry for the back and forth on this - I wasn't sure of your preferred error handling way of handling the `npf_mk_table()` failure scenario.